src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Match.cs

```diff
@@ -87,7 +88,216 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
             }
 
             string regularExpression = nodeValue;
-            return fValid && TryCreateReturnType(regExNode, regularExpression, errors, ref returnType);
+            return fValid && 
+                    (!context.Features.PowerFxV1CompatibilityRules || LimitRegularExpression(regExNode, regularExpression, errors)) &&
+                    TryCreateReturnType(regExNode, regularExpression, errors, ref returnType);
+        }
+
+        // Limit regular expressions to common features that are supported, with conssitent semantics, by both canonical .NET and XRegExp.
+        // It is better to disallow now and bring back with customer demand or as platforms add more support.
+        //
+        // Features that are disallowed:
+        //     Capture groups
+        //         Self-referncing groups, such as "(a\1)".
+        //         Treat all escaped number sequences as a backreference number.
+        //         Single quoted "(?'name'..." and "\k'name'".
+        //         Balancing capture groups.
+        //     Octal character codes (use Hex or Unicode instead).
+        //         "\o" could be added in the future, but we should avoid "\0" which causes backreference confusion.
+        //     Inline options
+        //         Anywhere in the expression except the beginning.
+        //         For subexpressions.
+        //     Character classes
+        //         Character class subtraction "[a-z-[m-n]]".
+        //     Conditional alternation
+        //
+        // Features that aren't supported by canonical .NET will be blocked automatically when the regular expression is instantiated in TryCreateReturnType.
+        //
+        // We chose to use canonical .NET instead of RegexOptions.ECMAScript because we wanted the unicode definitions for words.
+        // See https://learn.microsoft.com/dotnet/standard/base-types/regular-expression-options#ecmascript-matching-behavior for more details
+        private bool LimitRegularExpression(TexlNode regExNode, string regexPattern, IErrorContainer errors)
```

Comment:
Suggestion: rename to 'IsSupportedRegularExpression'; the name "LimitRegex" doesn't indicate whether it would return true or false in a valid / invalid one.

Reply:
Done.